cypress/integration/core_opensearch_dashboards/opensearch_dashboards/apps/explore/04/keyboard_shortcuts_dashboard.spec.js

```diff
@@ -0,0 +1,276 @@
+
+          cy.getElementByTestId('savedObjectSaveModal').should('be.visible');
+          cy.contains('Save dashboard').should('be.visible');
+
+          cy.get('body').type('{esc}');
+          cy.wait(500);
+        });
+      });
+
+      it('should open add panel flyout when pressing A key', () => {
+        cy.getElementByTestId('dashboardEditSwitch').then(($switch) => {
+          const isInEditMode = $switch.attr('aria-checked') === 'true';
+          if (!isInEditMode) {
+            cy.get('body').type('{shift+e}');
+            cy.wait(1000);
+          }
+
+          cy.get('body').type('a');
+          cy.wait(1000);
+          cy.getElementByTestId('dashboardAddPanel').should('be.visible');
+          cy.contains('Add panels').should('be.visible');
+
+          cy.get('body').type('{esc}');
+          cy.wait(500);
+        });
+      });
+    });
+
+    describe('Toggle Full-Screen Shortcuts', () => {
+      it('should enter full-screen mode when pressing Shift+F', () => {
+        cy.get('body').type('{shift+f}');
+        cy.wait(1000);
+
+        cy.getElementByTestId('exitFullScreenModeLogo').should('be.visible');
+        cy.contains('Exit full screen').should('be.visible');
+        cy.wait(1000);
+        cy.get('body').type('{esc}');
+        cy.wait(500);
+      });
+    });
+
+    describe('Refresh Results Shortcut', () => {
+      it('should trigger refresh action when pressing R key', () => {
+        cy.contains('.euiSuperUpdateButton__text', 'Refresh').should('be.visible');
+        cy.get('body').type('r');
+        cy.wait(1000);
+        cy.get('body').should('be.visible');
+        cy.contains('.euiSuperUpdateButton__text', 'Refresh').should('be.visible');
+        cy.contains('.euiSuperUpdateButton__text', 'Refresh').parent().should('not.be.disabled');
+      });
+    });
+
+    describe('Date Picker Shortcuts', () => {
+      it('should open date picker when pressing Shift+D', () => {
+        cy.get('body').then(($body) => {
+          if ($body.find('[data-test-subj="superDatePickerShowDatesButton"]').length > 0) {
+            cy.getElementByTestId('superDatePickerShowDatesButton').should('be.visible');
+            cy.get('body').type('{shift+d}');
+            cy.wait(1000);
+            cy.getElementByTestId('superDatePickerRelativeTab').should('be.visible');
+          }
+        });
+      });
+    });
+
+    describe('Focus Query Bar Shortcut', () => {
+      it('should focus query input textarea when pressing / key', () => {
+        cy.wait(3000);
+        cy.get('[data-test-subj="queryInput"]').should('be.visible');
+        cy.get('body').click();
+        cy.wait(100);
+        cy.get('[data-test-subj="queryInput"]').should('not.be.focused');
+        cy.get('body').type('/');
+        cy.wait(500);
+        cy.get('[data-test-subj="queryInput"]').should('be.focused');
+      });
+    });
+  });
+};
+
+prepareTestSuite('Dashboard Keyboard Shortcuts with Sample Data', runKeyboardShortcutsTests);
```

Comment:
E2E testing is different from unit testing. We don't want to go into too details into our testing in e2e but just ensure that the critical things are working.

There is another balancing act we gotta do. Our CI/CD takes an extremely long time due to cypress tests, and we dont' want to add too much time to that. So I'm going to propose several thigns:

1. make all the keyboard shortcut tests into one file (instead of three)
2. Things like "opening help menu", "checking content", "closing" should be handled in one `it` clause
3. group other keyboard shortcuts together, have one `it` test multiple in a single page